pyLucXor/psm.py

```
@@ -0,0 +1,1367 @@
+            bool: Whether the permutation is valid
+        """
+        try:
+            # Get unmodified sequence length
+            def get_unmodified_length(seq: str) -> int:
+                length = 0
+                i = 0
+                while i < len(seq):
+                    if seq[i:i+9] == "(Phospho)":
+                        i += 9
+                    elif seq[i:i+9] == "(Oxidation)":
+                        i += 9
```

Ruff (0.12.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)